### ISSUE TITLE
Uploading photo's - spinning circle under profile

### ISSUE THREAD

**Issue by Derilium10** [chiefonboarding/ChiefOnboarding#92]
Seem to be struggling uploading photos for the employees, i have a spinning circle.

Im a bit new to docker so be gentle...

**Comment by GDay** [comment]
You might have not added the credentials for S3. This app doesn't use the server for storage of files, this needs to be done through S3.

See the documentation for that: https://docs.chiefonboarding.com/integrations/S3.html
You will need to add those variables to your docker-compose file, under `environment` of the `web` part.

I will leave this ticket open to remind myself that I need to add an error message for that.

**Comment by Derilium10** [comment]
That makes sense, i was hoping to host everything myself, is there anyway i can get it to save the documents within it or maybe on a system (own-hosted)

**Comment by GDay** [comment]
You will have to use an external application/vendor for this. Though, if you want to self-host, then you could look into [minio](https://github.com/minio/minio). It can be ran as a docker container. I have used that before and it's basically a local S3 equivalent. ChiefOnboarding doesn't need AWS S3, it should be able to work with basically any service that provides an S3 compatible API.

The reason why you cannot upload to the app directly, is because it would open the app up to potential DDoS attacks. I would have to extend the amount of minutes that a connection can stay open, because some files might take a long time to upload. So, currently, all uploads are done directly to object storage.

**Comment by Kuuchuu** [comment]
@GDay How would one go about configuring the enviroment variables for minio? As far as I can tell everything is set up how I believe it should be. Photos still won't upload.
Environment Variables:
```
      - AWS_S3_ENDPOINT_URL=http://chiefonboarding_minio:9000
      - AWS_ACCESS_KEY_ID=***********
      - AWS_SECRET_ACCESS_KEY=*********
      - AWS_STORAGE_BUCKET_NAME=chiefonboarding
      - AWS_REGION=db-1
```
"chiefonboarding_minio" is the name of the service and container for minio, they are both in the same compose file and in the same network.
"db-1" is set as the minio location in its console. The bucket has been created.
Minio has the MINIO_API_CORS_ALLOW_ORIGIN environment variable configured.

Dev Console:
```
2a05b5d.js:2
POST https://workfor.anchorhrg.com/api/org/file 500
(anonymous) @ 2a05b5d.js:2
t.exports @ 2a05b5d.js:2
t.exports @ 2a05b5d.js:2
Promise.then (async)
h.request @ 2a05b5d.js:2
r.forEach.h.<computed> @ 2a05b5d.js:2
(anonymous) @ 2a05b5d.js:2
getPreSignedURL @ 04f882c.js:1
change @ 04f882c.js:1
te @ 2a05b5d.js:2
n @ 2a05b5d.js:2
Lr.c._wrapper @ 2a05b5d.js:2
```
```
04f882c.js:1
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'data')
    at 04f882c.js:1:376286
(anonymous) @ 04f882c.js:1
Promise.then (async)
change @ 04f882c.js:1
te @ 2a05b5d.js:2
n @ 2a05b5d.js:2
Lr.c._wrapper @ 2a05b5d.js:2
```
Thanks in advance!

**Comment by GDay** [comment]
@Kuuchuu Your mistake it probably using an internal url for the endpoint. You can't do that unless you are in that same network (via a vpn or whatever). 

I just tested it up on a VPS and here is the config I used (I was able to post to Minio without issues):

`docker-compose.yaml`
```
version: '3'                  
                                                                                                                   
services:                      
  db:                                   
    image: postgres:latest                 
    restart: always         
    expose:    
      - "5432"
    volumes: 
      - pgdata:/var/lib/postgresql/data/
    environment:
      - POSTGRES_DB=chiefonboarding
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
    networks:
      - global 
                                                         
  web:      
    image: chiefonboarding/chiefonboarding:latest
    restart: always   
    expose:             
      - "8000"              
    environment:
      - SECRET_KEY=somethingsupersecret
      - BASE_URL=https://test.chiefonboarding.com
      - DATABASE_URL=postgres://postgres:postgres@db:5432/chiefonboarding
      - ALLOWED_HOSTS=test.chiefonboarding.com
      - DEFAULT_FROM_EMAIL=hello@example.com
      - ACCOUNT_EMAIL=hello@example.com
      - ACCOUNT_PASSWORD=password
      - AWS_S3_ENDPOINT_URL=https://minio.chiefonboarding.com
      - AWS_ACCESS_KEY_ID=chief
      - AWS_SECRET_ACCESS_KEY=chiefpass
      - AWS_STORAGE_BUCKET_NAME=test-bucket
      - AWS_REGION=us-east-1
    depends_on:                                      
      - db                                                                                                         
    networks:
      - global

  caddy:
    image: caddy:2.3.0-alpine
    restart: unless-stopped
    ports:     
      - "80:80"
      - "443:443"
    volumes:
      - $PWD/Caddyfile:/etc/caddy/Caddyfile
      - $PWD/site:/srv
      - caddy_data:/data
      - caddy_config:/config
    networks:
      - global

  minio-server:
    image: minio/minio
    expose:
      - "9000"
      - "9001"
    volumes:
      - ./storage/minio:/data
    environment:
      - MINIO_ROOT_USER=chief
      - MINIO_ROOT_PASSWORD=chiefpass
      - MINIO_DOMAIN=http://minio.chiefonboarding.com
    command: server --address 0.0.0.0:9000 --console-address 0.0.0.0:9001 /data
    networks:
      - global

volumes:
  pgdata:
  caddy_data:
  caddy_config:

networks:
  global:
```

`Caddyfile`:
```
test.chiefonboarding.com {
  reverse_proxy web:8000
}
minio.chiefonboarding.com {
  reverse_proxy minio-server:9000
}
minio-console.chiefonboarding.com {
  reverse_proxy minio-server:9001
}
```

After setting it up, I went to the minio console and created a bucket there. I didn't change the region and therefore I had to put `us-east-1` there.

As a side note: you left your domainname in the error message, not sure if that was intended.
Also, for some reason, I didn't have to set CORS on minio. It just worked out of the box.

**Comment by GDay** [comment]
Closing this as I have released v2.0 and it now clearly shows an error message when object storage has not been set up: 

![image](https://user-images.githubusercontent.com/1939656/178082661-e040b4be-79ed-4f47-ae27-34f3680828b5.png)
